_posts/2023-04-04-segment-replication.md

```diff
@@ -0,0 +1,249 @@
+
+## Document replication
+
+For versions before 2.7, OpenSearch propagates source documents from the primary to the replicas for indexing. All operations that affect an OpenSearch index (for example, adding, updating, or removing documents) are routed to the index’s primary shards. The primary shard is responsible for validating the operation and subsequently executing it locally. Once the operation has been completed successfully on the primary shard, the operation is forwarded to each of its replica shards in parallel. Each replica shard executes the forwarded operation, duplicating the processing performed on the primary shard. When the operation has completes (either successfully or with a failure) on every replica and a response has been received by the primary shard, the primary shard responds to the client. The response includes information about replication success or failure on replica shards.
+
+The advantage of document replication is that documents are immediately sent to the replicas, where they become searchable as. The system reaches a consistent state between primaries and replicas as quickly as possible. But, because the indexing operation happens _n_ times (primary + replica count) for each document, document replication consumes more CPU.
+
+Refer to the following diagram of the document replication process.
+
+<img src="/assets/media/blog-images/2023-04-04-segment-replication/document-replication.png" alt="Document replication diagram"/>{: .img-fluid}
+
+## Segment replication
+
+With segment replication, documents are indexed only on the node that contains the primary shard. The produced segment files are then copied directly to all replicas and are made searchable. Segment replication reduces the compute cost for adding, updating, or deleting documents by doing the CPU work only on the nodes with the primary shards. The underlying Lucene append-only indexing makes copying segments possible: as documents are added, updated, or deleted, Lucene creates new segments, but the existing segments are left untouched (deletes are handled with tombstones).
+
+The advantage of segment replication is that it reduces the CPU usage overall in your cluster by removing the duplicated effort of parsing and processing the data in your documents. But, because all indexing and networking originates on the nodes with primary shards, those nodes become more heavily loaded. Also, nodes with primary shards spend time waiting for segment creation (controlled by the _\_refresh\_interval_) and time sending the segments to the replica, increasing the time before a particular document is consistently searchable on every shard.
+
+You [enable segment replication](https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/segment-replication/configuration/) for your cluster and then per index. Refer to the following diagram of the segment replication process.
```

Comment:
"for your cluster" 

This is not required - we are looking at a cluster level setting that is currently behind a feature flag.

With GA all that is required is to enable at index creation.